### ISSUE TITLE
Blazor Wasm: published app broken possibly from linking

### ISSUE THREAD

**Issue by zbecknell** [dotnet/aspnetcore#24391]
### Describe the bug
.NET 5 preview 7:

App works when using `dotnet run` (regardless of configuration) but does not work when publishing via `dotnet publish`.  This seems to possibly be linker related.  I can't tell that `<BlazorWebAssemblyEnableLinking>` has any effect in preview7.  Is there a new way of disabling linking entirely?

The exception (in our main app, it's different in the repro but still the same pattern):

```
An exception occurred while creating the WASM host 
 System.MissingMethodException: Default constructor not found for type System.ComponentModel.Int32Converter
   at System.RuntimeType.CreateInstanceMono(Boolean nonPublic, Boolean wrapExceptions)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean skipCheckThis, Boolean fillCache)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic)
   at System.Activator.CreateInstance(Type type)
   at System.ComponentModel.ReflectTypeDescriptionProvider.CreateInstance(Type objectType, Type callingType)
   at System.ComponentModel.ReflectTypeDescriptionProvider.SearchIntrinsicTable(Hashtable table, Type callingType)
   at System.ComponentModel.ReflectTypeDescriptionProvider.ReflectedTypeData.GetConverter(Object instance)
   at System.ComponentModel.ReflectTypeDescriptionProvider.GetConverter(Type type, Object instance)
   at System.ComponentModel.TypeDescriptor.TypeDescriptionNode.DefaultTypeDescriptor.System.ComponentModel.ICustomTypeDescriptor.GetConverter()
   at System.ComponentModel.TypeDescriptor.GetConverter(Type type)
   at AutoMapper.Mappers.TypeConverterMapper.GetTypeConverter(Type type)
   at AutoMapper.Mappers.TypeConverterMapper.IsMatch(TypePair context)
   at AutoMapper.MapperConfiguration.<>c__DisplayClass72_0.<FindMapper>b__0(IObjectMapper m)
   at System.Linq.Enumerable.TryGetFirst[IObjectMapper](IEnumerable`1 source, Func`2 predicate, Boolean& found)
   at System.Linq.Enumerable.FirstOrDefault[IObjectMapper](IEnumerable`1 source, Func`2 predicate)
   at AutoMapper.MapperConfiguration.FindMapper(TypePair types)
   at AutoMapper.Execution.TypeMapPlanBuilder.<CheckForCycles>g__ResolveTypeMap|15_6(TypePair types)
   at AutoMapper.Execution.TypeMapPlanBuilder.<CheckForCycles>g__ResolveMemberTypeMap|15_5(IMemberMap memberMap)
   at AutoMapper.Execution.TypeMapPlanBuilder.<CheckForCycles>b__15_2(IMemberMap pm)
   at System.Linq.Enumerable.SelectArrayIterator`2[[AutoMapper.IMemberMap, AutoMapper, Version=10.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005],[<>f__AnonymousType4`2[[AutoMapper.TypeMap, AutoMapper, Version=10.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005],[AutoMapper.IMemberMap, AutoMapper, Version=10.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005]], AutoMapper, Version=10.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005]].MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1[[<>f__AnonymousType4`2[[AutoMapper.TypeMap, AutoMapper, Version=10.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005],[AutoMapper.IMemberMap, AutoMapper, Version=10.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005]], AutoMapper, Version=10.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005]].MoveNext()
   at AutoMapper.Execution.TypeMapPlanBuilder.CheckForCycles(HashSet`1 typeMapsPath)
   at AutoMapper.Execution.TypeMapPlanBuilder.CreateMapperLambda(HashSet`1 typeMapsPath)
   at AutoMapper.TypeMap.CreateMapperLambda(IConfigurationProvider configurationProvider, HashSet`1 typeMapsPath)
   at AutoMapper.TypeMap.Seal(IConfigurationProvider configurationProvider)
   at AutoMapper.MapperConfiguration.Seal()
   at AutoMapper.MapperConfiguration..ctor(MapperConfigurationExpression configurationExpression)
   at AutoMapper.MapperConfiguration..ctor(Action`1 configure)
   at Care.Client.Program.Main(String[] args)
```
 
<img width="1466" alt="image" src="https://user-images.githubusercontent.com/9710256/88744460-6b200380-d115-11ea-977f-8a6870a91ded.png">

### To Reproduce
See reproduction project here: https://github.com/zbecknell/Net5Preview7Repro

Use `dotnet run` and all is well.  Use the output from `dotnet publish` and the app bombs when starting up.

### Further technical details

- Include the output of `dotnet --info`

```
.NET SDK (reflecting any global.json):
 Version:   5.0.100-preview.7.20366.6
 Commit:    0684df3a5b
Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-preview.7.20366.6\
Host (useful for support):
  Version: 5.0.0-preview.7.20364.11
  Commit:  53976d38b1
.NET SDKs installed:
  3.1.103 [C:\Program Files\dotnet\sdk]
  3.1.202 [C:\Program Files\dotnet\sdk]
  3.1.301 [C:\Program Files\dotnet\sdk]
  3.1.400-preview-015203 [C:\Program Files\dotnet\sdk]
  5.0.100-preview.7.20366.6 [C:\Program Files\dotnet\sdk]
.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-preview.7.20365.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-preview.7.20364.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-preview.7.20366.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by zbecknell** [comment]
Some things I attempted:

* `<BlazorWebAssemblyEnableLinking>false</...>` -> same error
* `<PublishTrimmed>false</...>` -> resource integrity (sha256) failure
* Tried a custom `LinkerConfig.xml` to avoid linking as much as I could, but didn't seem to have an effect on the bug.

Perhaps linking is a red herring and there is something else happening.  I just know that whatever effects take place during `publish` cause this issue.

For now we are still up and running, as we dual-deploy SSB and CSB, so our SSB app is still functioning (thanks to the versatility of Blazor!).  But if there is a suggested workaround to avoid the effects of `dotnet publish` I'd be grateful to hear it!

**Comment by zbecknell** [comment]
Ok, I got it working.  First it worked locally by setting `-p:PublishTrimmed=false` but would still fail when deployed via hashes not matching.

My deployment process publishes and then zips the resulting output.  This zipped output showed both `.br` and `.gz` files that were smaller than the raw (unzipped) output once extracted.  Perhaps these already compressed formats have trouble with being zipped in this manner.

I'm guessing linking is in a state of flux right now and things will calm down with subsequent previews.

**Comment by pranavkm** [comment]
Likely a dup of dotnet/runtime#39973 which is fixed in 5.0-rc1.
